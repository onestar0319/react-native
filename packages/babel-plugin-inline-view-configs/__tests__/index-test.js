/**
 * Copyright (c) Facebook, Inc. and its affiliates.
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 *
 * @emails oncall+react_native
 * @format
 */

'use strict';

const {transform: babelTransform} = require('@babel/core');
const fixtures = require('../__test_fixtures__/fixtures.js');

function transform(filename) {
  return babelTransform(fixtures[filename], {
    plugins: [require('@babel/plugin-syntax-flow'), require('../index')],
    babelrc: false,
    filename,
  }).code;
}

describe('Babel plugin inline view configs', () => {
  Object.keys(fixtures)
    .sort()
    .forEach(fixtureName => {
      it(`can inline config for ${fixtureName}`, () => {
        expect(transform(fixtureName)).toMatchSnapshot();
      });
    });
});
