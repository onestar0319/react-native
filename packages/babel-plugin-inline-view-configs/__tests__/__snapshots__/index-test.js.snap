// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Babel plugin inline view configs can inline config for FullNativeComponent.js 1`] = `
"// @flow
const codegenNativeCommands = require('codegenNativeCommands');

const codegenNativeComponent = require('codegenNativeComponent');

import type { Int32, BubblingEventHandler, DirectEventHandler, WithDefault } from 'CodegenFlowtypes';
import type { ViewProps } from 'ViewPropTypes';
interface NativeCommands {
  +hotspotUpdate: (viewRef: React.Ref<'RCTView'>, x: Int32, y: Int32) => void,
  +scrollTo: (viewRef: React.Ref<'RCTView'>, y: Int32, animated: boolean) => void,
}
type ModuleProps = $ReadOnly<{| ...ViewProps,
  // Props
  boolean_default_true_optional_both?: WithDefault<boolean, true>,
  // Events
  onDirectEventDefinedInlineNull: DirectEventHandler<null>,
  onBubblingEventDefinedInlineNull: BubblingEventHandler<null>,
|}>;

const registerGeneratedViewConfig = require('registerGeneratedViewConfig');

const {
  UIManager
} = require(\\"react-native\\");

const {
  findNodeHandle
} = require(\\"react-native\\");

const ModuleViewConfig = {
  uiViewClassName: 'RCTModule',
  bubblingEventTypes: {
    topBubblingEventDefinedInlineNull: {
      phasedRegistrationNames: {
        captured: 'onBubblingEventDefinedInlineNullCapture',
        bubbled: 'onBubblingEventDefinedInlineNull'
      }
    }
  },
  directEventTypes: {
    topDirectEventDefinedInlineNull: {
      registrationName: 'onDirectEventDefinedInlineNull'
    }
  },
  validAttributes: {
    boolean_default_true_optional_both: true,
    onDirectEventDefinedInlineNull: true,
    onBubblingEventDefinedInlineNull: true
  }
};
let nativeComponentName = 'RCTModule';
registerGeneratedViewConfig(nativeComponentName, ModuleViewConfig);
export const __INTERNAL_VIEW_CONFIG = ModuleViewConfig;
export default nativeComponentName;
export const Commands = {
  hotspotUpdate(ref, x, y) {
    UIManager.dispatchViewCommand(findNodeHandle(ref), UIManager.getViewManagerConfig(\\"RCTModule\\").Commands.hotspotUpdate, [x, y]);
  },

  scrollTo(ref, y, animated) {
    UIManager.dispatchViewCommand(findNodeHandle(ref), UIManager.getViewManagerConfig(\\"RCTModule\\").Commands.scrollTo, [y, animated]);
  }

};"
`;

exports[`Babel plugin inline view configs can inline config for NotANativeComponent.js 1`] = `
"const requireNativeComponent = require('requireNativeComponent');

export default 'Not a view config';"
`;

exports[`Babel plugin inline view configs fails on inline config for CommandsExportedWithDifferentNameNativeComponent.js 1`] = `"Native commands must be exported with the name 'Commands'"`;

exports[`Babel plugin inline view configs fails on inline config for CommandsExportedWithShorthandNativeComponent.js 1`] = `"'Commands' is a reserved export and may only be used to export the result of codegenNativeCommands."`;

exports[`Babel plugin inline view configs fails on inline config for OtherCommandsExportNativeComponent.js 1`] = `"'Commands' is a reserved export and may only be used to export the result of codegenNativeCommands."`;
